### PR TITLE
Fix race condition for some threading tests using GC.KeepAlive

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @sergiy-k, @jkotas @RussKeldorph 

I observed that these tests were failing in .NET Native under GC stress.

### CODE REVIEW

File: tests/src/baseservices/threading/waithandle/waitany/nullarray.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ static int Main()
 			retCode = 98;			
 		}
 
+		GC.KeepAlive(waitHandles);
```
Comment (sergiy-k) [inline tests/src/baseservices/threading/waithandle/waitany/nullarray.cs]: Hmm... I'm not sure why you need this. ```waitHandles``` is always set to null.

Comment (RussKeldorph) [inline tests/src/baseservices/threading/waithandle/waitany/nullarray.cs]: I agree.  What was the observed behavior of this test under GCStress without this change?

---
In reply to: [132806971](https://github.com/dotnet/coreclr/pull/13351#discussion_r132806971) [](ancestors = 132806971)

Comment (ahsonkhan) [inline tests/src/baseservices/threading/waithandle/waitany/nullarray.cs]: Yes, you are correct. I did not intend to check in this change (I have reverted it).

The fix for the null array test failure was fixed in corert here: https://github.com/dotnet/corert/pull/4312

**Comment by ahsonkhan** [comment]
Good to go?

@sergiy-k, @RussKeldorph

**Pull request merged**